server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -951,7 +951,7 @@ private void checkForTimeout(long requestId) {
         assert responseHandlers.contains(requestId) == false;
         TimeoutInfoHolder timeoutInfoHolder = timeoutInfoHandlers.remove(requestId);
         if (timeoutInfoHolder != null) {
-            long time = System.currentTimeMillis();
+            long time = threadPool.absoluteTimeInMillis();
```

Comment:
I'm not sure these changes should be hidden away in this PR. I also think these should rather be relativeTimeMillis as we are only interested in deltas here. Maybe a separate PR on master?

Reply:
I opened #33869 and changed the calls here to use `relativeTimeMills` in f69f20e37775bf0ade30caeb5c87adba02ada7d9.